ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -755,6 +753,13 @@ int closeInProcess(boolean destroyed) {
   // is called to stop the query if it is running, clean query results, and release resources.
   @Override
   public void close() {
+    if (driverContext.getConf().getBoolVar(HiveConf.ConfVars.HIVE_QUERY_HISTORY_SERVICE_ENABLED)) {
+      if (driverState.isClosed() || driverState.isDestroyed()){
+        LOG.warn("Driver instance {} already closed or destroyed, prevent handling query history", this);
+      } else{
+        QueryHistoryService.handle(driverContext);
```

Comment:
the method name is wage, what does it do `handle`?

Reply:
this is the main entrypoint of the service, where it gets the driverContext, and handles it (convert to QueryHistoryRecord, queue to persist), what name would make more sense here?

Reply:
renamed to handleQuery, I don't know better at the moment

Reply:
(add/register/dump/record)Query